### ISSUE TITLE
App breaks on switching to another story

### ISSUE THREAD

**Issue by irudoy** [storybookjs/storybook#18011]
**Describe the bug**
App breaks in runtime when navigating to some stories.

Storybook v6.5.0-alpha.64 It worked fine on v6.4.22.

```
265.manager.bundle.js:53840
Uncaught DOMException: Failed to execute 'removeChild' on 'Node': The node to be removed is not a child of this node.
    at removeChild (http://localhost:6006/265.manager.bundle.js:42245:18)
    at unmountHostComponents (http://localhost:6006/265.manager.bundle.js:55102:9)
    at commitDeletion (http://localhost:6006/265.manager.bundle.js:55161:5)
    at commitMutationEffects_begin (http://localhost:6006/265.manager.bundle.js:55373:11)
    at commitMutationEffects (http://localhost:6006/265.manager.bundle.js:55357:3)
    at commitRootImpl (http://localhost:6006/265.manager.bundle.js:57808:5)
    at commitRoot (http://localhost:6006/265.manager.bundle.js:57687:5)
    at performSyncWorkOnRoot (http://localhost:6006/265.manager.bundle.js:57126:3)
    at flushSyncCallbacks (http://localhost:6006/265.manager.bundle.js:43157:22)
    at flushSyncCallbacksOnlyInLegacyMode (http://localhost:6006/265.manager.bundle.js:43136:5)
```

**To Reproduce**
1) Go to https://dromru.github.io/react-photoswipe-gallery/storybook-storysource/
2) Click on `DEMO -> Custom Content` or `DEV -> Playground` in the sidebar

https://github.com/dromru/react-photoswipe-gallery/tree/storybook-storysource

**System**
  System:
    OS: macOS 12.2.1
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
  Binaries:
    Node: 17.8.0 - /usr/local/bin/node
    Yarn: 3.2.0 - /usr/local/bin/yarn
    npm: 8.5.5 - /usr/local/bin/npm
  Browsers:
    Chrome: 102.0.4997.0
    Safari: 15.3

**Comment by irudoy** [comment]
Seems like it's also related to https://github.com/storybookjs/storybook/issues/18012 (addon-storysource)
This issue occurs only when the active tab is a `Story` which is broken. It works fine with another tab, e. g. `Controls`.

**Comment by irudoy** [comment]
Still actual for v6.5.0-beta.0

**Comment by irudoy** [comment]
Fixed in v6.5.0-beta.7
